### PR TITLE
Embeddable games

### LINKED ISSUE #893: Support embedding OGS on other sites with iframe html-tags
If we support to link OGS games in iframes, it would be possible to embed OGS games on other pages (the forum for example). This could serve as a replacement of the abandoned sgf viewer of the forum. 

There would be the possibility of others using it on their own sites as well.

As a very basic idea of how it could work see lichess: https://lichess.org/developers

The iframe version could look like this for example:
iframe-example1

----

The iframe support could either happen by using separate URLs or by detecting if OGS is embedded in JavaScript. 

A simple version of this would be to simply hiding the NavBar, right-col, disable sounds.
In a more enhanced version we could add the possibility to select a move/variation to be displayed with URL parameters for example.

If you are interested, I'll try to implement iframe support and a discourse plugin for the forum.

### PR DESCRIPTION
**Pull request by benjaminpjones**
Fixes #893 

## Proposed Changes

  - Put a MiniGoban at online-go.com/game/%%/embed
  - Add sample iframe code to the game links

Screen Shot 2022-06-05 at 1 38 02 PM
Screen Shot 2022-06-05 at 1 40 02 PM

### CODE REVIEW

**Comment by benjaminpjones** [comment]
Drats, my attempts to delete goban have been foiled!

**Review by anoek — COMMENTED** [review]
Pretty slick!

@Fraze is this feature still useful to you, and if so, how many embeddable games were you envisioning per page? Right now it basically has a live updating game that is clickable, is that about what you had in mind?

Fraze's use case might be a little different than @flovo's too. I think flovo had in mind some basic controls and whatnot so one could at least skip around in the game to view different moves and whatnot, akin to an embeddable sgf viewer or something except for ogs games - but that seems perhaps a little orthogonal to what Fraze might want which would be (I think) links to games as you have it now, and probably not want a bunch of controls to confuse the match making situation.

File: src/components/MiniGoban/MiniGoban.tsx [hunk 1]
```diff
@@ -232,6 +233,14 @@ export function MiniGoban(props: MiniGobanProps): JSX.Element {
         };
     }, [props.id]);
 
+    // Update displayWidth dynamically
+    React.useEffect(() => {
+        if (!goban.current && props.displayWidth == null) {
```
[the remark this answers is not in the document]
Comment (benjaminpjones) [reply src/components/MiniGoban/MiniGoban.tsx]: Yikes, good catch!

**Comment by benjaminpjones** [comment]
Thanks!

I did speak a bit on this with @fraze before implementing, and it sounded like the need for embeddable games had kind of passed. But yeah I think this implementation hits a little closer to @fraze's needs (and [this streamer's](https://forums.online-go.com/t/embed-live-game/42216/6)). @fraze's use-case can be found here: https://baduk.club/mac/2022va-open, just click **Board 1** to show the `iframe`.

>an embeddable sgf viewer or something except for ogs games

I'd find this useful too.  I think we have a few options to add this eidogo-style interface in the future:

- eidogo-style for review/demo embeds only (this PR does not implement `/review/%%/embed`)
    - If somebody wants to have an analyzable game, it is trivial to create a review from a game.
- check the iframe dimensions: if there is significant horizontal or vertical space, include analysis tools
- query params: something like `interactive=true` or `analysis=true`

Unfortunately, I don't think `MiniGoban` is a good fit for any of this stuff, but a lot of my futzing in `Game.tsx` has been with embeds in mind, so it shouldn't be hard to re-use some components from there. It was only at the last moment that I realized `MiniGoban` might be best for the default case.

**Comment by anoek** [comment]
Sounds good, and honestly for the eidogo style embeds I kinda always wanted to evolve the `goban` library to be capable of fulfilling that role anyways so maybe it's best to keep things as they are here, and then address the more complicated use case  with `goban` someday.

**Pull request merged**